tests/Microsoft.IdentityModel.Protocol.Extensions.Tests/CrossTokenTests.cs

```diff
@@ -70,9 +70,46 @@ public void CrossToken_ValidateToken()
 
+            SecurityKey x509SecurityKey = new X509SecurityKey(KeyingMaterial.DefaultCert_2048);
+            SigningCredentials signingCredentials = new SigningCredentials(x509SecurityKey, SecurityAlgorithms.RsaSha256Signature, SecurityAlgorithms.Sha256Digest);
+            SecurityTokenDescriptor descriptor = new SecurityTokenDescriptor
+            {
+                AppliesToAddress = IdentityUtilities.DefaultAudience,
+                SigningCredentials = signingCredentials,
+                Subject = IdentityUtilities.DefaultClaimsIdentity,
+                TokenIssuerName = IdentityUtilities.DefaultIssuer,
+                Lifetime = new System.IdentityModel.Protocols.WSTrust.Lifetime(DateTime.UtcNow, DateTime.UtcNow + TimeSpan.FromDays(1)),
+            };
+
+            TokenValidationParameters validationParameters = new TokenValidationParameters
+            {
+                AuthenticationType = IdentityUtilities.DefaultAuthenticationType,
+                IssuerSigningKey = x509SecurityKey,
+                ValidAudience = IdentityUtilities.DefaultAudience,
+                ValidIssuer = IdentityUtilities.DefaultIssuer,
+                ValidateIssuerSigningKey = true
+            };
+
+            // Test ValidateIssuerSigningKey = true, validationParameters.CertificateValidator == null, validationParameters.IssuerSigningKeyValidator == null
+            string jwtTokenTest = IdentityUtilities.CreateJwtToken(descriptor, jwtHandler);
+            ClaimsPrincipal jwtPrincipalTest = ValidateToken(jwtTokenTest, validationParameters, jwtHandler, ExpectedException.SecurityTokenValidationException("IDX10232:"));
+
+            // Test ValidateIssuerSigningKey = true, validationParameters.CertificateValidator == null, validationParameters.IssuerSigningKeyValidator != null
+            validationParameters.IssuerSigningKeyValidator = IssuerSigningKeyValidator;
+            jwtPrincipalTest = ValidateToken(jwtTokenTest, validationParameters, jwtHandler, ExpectedException.NoExceptionExpected);
+
             string nullIssuerJwtToken = IdentityUtilities.CreateJwtToken(IdentityUtilities.NullIssuerAsymmetricSecurityTokenDescriptor, jwtHandler);
             string jwtToken = IdentityUtilities.CreateJwtToken(IdentityUtilities.DefaultAsymmetricSecurityTokenDescriptor, jwtHandler);
 
+            // Test ValidateIssuerSigningKey = true, signingkey is not X509SecurityKey, alidationParameters.IssuerSigningKeyValidator == null
+            validationParameters = IdentityUtilities.DefaultAsymmetricTokenValidationParameters;
+            validationParameters.ValidateIssuerSigningKey = true;
+            jwtPrincipalTest = ValidateToken(jwtToken, validationParameters, jwtHandler, ExpectedException.SecurityTokenValidationException("IDX10233:"));
+
+            // Test ValidateIssuerSigningKey = true, signingkey is not X509SecurityKey, alidationParameters.IssuerSigningKeyValidator != null
```

Comment:
It's good to explicitly set each field for each test. That way, if someone else adds tests in between yours, they won't break your tests.